src/badge/withBadge.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import Badge from './Badge';
+
+const withBadge = (value, options = {}) => WrappedComponent => {
+  class WithBadge extends React.Component {
+    render() {
+      const {
+        left = 0,
+        bottom = 0,
+        top = -5,
+        right = 0,
+        hidden = false,
+        ...badgeProps
+      } = options;
+
+      const badgeValue =
+        typeof value === 'function' ? value(this.props) : value;
+
+      return (
+        <View>
```

Comment:
I think we may need to add some styling here. There's a weird display bug.

Generated by: 
```jsx
const BadgedAvatar = withBadge(1)(Avatar);

<View style={styles.container}>
  <BadgedAvatar
    size="large"
    source={{ uri: 'https://randomuser.me/api/portraits/men/44.jpg' }}
  />
</View>

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 30,
    paddingTop: Constants.statusBarHeight,
    // alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'hsl(116, 10%, 98%)',
  },
});
```
![simulator screen shot - iphone x - 2018-12-07 at 11 26 12](https://user-images.githubusercontent.com/5962998/49656135-ec8fc300-fa12-11e8-8a72-da2d26081bb1.png)

Styling on the outer view maybe?
```suggestion
        <View style={{ position: 'relative', alignItems:'center', justifyContent: 'center' }}>
```

![simulator screen shot - iphone x - 2018-12-07 at 11 29 38](https://user-images.githubusercontent.com/5962998/49656312-6a53ce80-fa13-11e8-8e6b-e50646634a7e.png)